Remove menuitem element and some leftover touchbar code

annevk opened the pull request:
#### 449b54f22d5ffc59b830838126d7e342e8748e0a
<pre>
Remove menuitem element and some leftover touchbar code
<a href="https://bugs.webkit.org/show_bug.cgi?id=260428">https://bugs.webkit.org/show_bug.cgi?id=260428</a>
rdar://114132124

Reviewed by Wenson Hsieh.

This code does not do anything and as such it is better to remove it to
ease ongoing maintenance.

This deprecates and no-ops WKPreferencesSetMenuItemElementEnabled and
WKPreferencesGetMenuItemElementEnabled in order to preserve binary
compatibility.

Due to file removal a couple unified build fixes were needed. Those are
annotated inline below.

* LayoutTests/fast/html/menuitem-element-expected.txt: Removed.
* LayoutTests/fast/html/menuitem-element.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/html/HTMLMenuElement.cpp:
(WebCore::HTMLMenuElement::insertedIntoAncestor): Deleted.
(WebCore::HTMLMenuElement::removedFromAncestor): Deleted.
(WebCore::HTMLMenuElement::attributeChanged): Deleted.
* Source/WebCore/html/HTMLMenuElement.h:
* Source/WebCore/html/HTMLMenuItemElement.cpp: Removed.
* Source/WebCore/html/HTMLMenuItemElement.h: Removed.
* Source/WebCore/html/HTMLMenuItemElement.idl: Removed.
* Source/WebCore/html/HTMLTagNames.in:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::didInsertMenuElement): Deleted.
(WebCore::ChromeClient::didRemoveMenuElement): Deleted.
(WebCore::ChromeClient::didInsertMenuItemElement): Deleted.
(WebCore::ChromeClient::didRemoveMenuItemElement): Deleted.
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.h:

Unified build fix.

* Source/WebKit/Shared/TouchBarMenuData.cpp: Removed.
* Source/WebKit/Shared/TouchBarMenuData.h: Removed.
* Source/WebKit/Shared/TouchBarMenuItemData.cpp: Removed.
* Source/WebKit/Shared/TouchBarMenuItemData.h: Removed.
* Source/WebKit/Shared/TouchBarMenuItemData.serialization.in: Removed.
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetMenuItemElementEnabled): Deleted.
(WKPreferencesGetMenuItemElementEnabled): Deleted.
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::touchBarMenuDataChanged): Deleted.
(WebKit::WebPageProxy::touchBarMenuItemDataAdded): Deleted.
(WebKit::WebPageProxy::touchBarMenuItemDataRemoved): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::didInsertMenuElement): Deleted.
(WebKit::WebChromeClient::didRemoveMenuElement): Deleted.
(WebKit::WebChromeClient::didInsertMenuItemElement): Deleted.
(WebKit::WebChromeClient::didRemoveMenuItemElement): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didInsertMenuElement): Deleted.
(WebKit::WebPage::didRemoveMenuElement): Deleted.
(WebKit::WebPage::didInsertMenuItemElement): Deleted.
(WebKit::WebPage::didRemoveMenuItemElement): Deleted.
(WebKit::WebPage::sendTouchBarMenuDataRemovedUpdate): Deleted.
(WebKit::WebPage::sendTouchBarMenuDataAddedUpdate): Deleted.
(WebKit::WebPage::sendTouchBarMenuItemDataAddedUpdate): Deleted.
(WebKit::WebPage::sendTouchBarMenuItemDataRemovedUpdate): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:

Includes unified build fix.

* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences menuItemElementEnabled]): Deleted.
(-[WebPreferences setMenuItemElementEnabled:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebPreferencesPrivate.h:

Canonical link: <a href="https://commits.webkit.org/267140@main">https://commits.webkit.org/267140@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8f60b0024faa6ef5c4680fde068e0e4fdb5fabe5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15792 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16110 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16489 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17554 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14822 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15977 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19110 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16207 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17317 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15982 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16457 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13449 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18309 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13706 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14261 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/21165 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/13578 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14701 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14426 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17686 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/15038 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15021 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12736 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/15990 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14268 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/4010 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3770 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18637 "Built successfully") | | [❌ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/17125 "Failed to checkout and rebase branch from PR 16864") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14843 "Built successfully") | | [❌ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/24/builds/17125 "Failed to checkout and rebase branch from PR 16864") | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.h:
```diff
@@ -54,7 +56,7 @@ class RemoteScrollingCoordinatorTransaction {
 
 #if !defined(NDEBUG) || !LOG_DISABLED
-    String description() const;
+    WTF::String description() const;
```
Hm...it's a tiny bit surprising that we'd need a `WTF::` here. Is that because it conflicts with `API::String` from a nearby unified source?

annevk replied:
I'm not sure, the compiler "asked" if I meant `std::string` instead. I can try again, maybe something changed meanwhile.

whsieh replied:
> the compiler "asked" if I meant `std::string` instead

Oh, I see — this is the "hint" part of the error message you'd get if you used `String` without including `WTFString.h` first; including `WTFString.h` above was sufficient to address this.

whsieh commented on Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
```diff
@@ -879,7 +879,7 @@ static void dump(TextStream& ts, const ScrollingStateTree& stateTree, bool chang
 }
 
-String RemoteScrollingCoordinatorTransaction::description() const
+WTF::String RemoteScrollingCoordinatorTransaction::description() const
```
(Ditto, re: `WTF::`)

whsieh commented on Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
```diff
@@ -1557,16 +1557,6 @@ bool WKPreferencesGetIntersectionObserverEnabled(WKPreferencesRef preferencesRef
     return toImpl(preferencesRef)->intersectionObserverEnabled();
 }
 
-void WKPreferencesSetMenuItemElementEnabled(WKPreferencesRef preferencesRef, bool flag)
```
It might be a good idea to keep these two SPI functions around as an empty stub, only for bincompat.

annevk committed:
Remove menuitem element and some leftover touchbar code
https://bugs.webkit.org/show_bug.cgi?id=260428
rdar://114132124

Reviewed by Wenson Hsieh.

This code does not do anything and as such it is better to remove it to
ease ongoing maintenance.

This deprecates and no-ops WKPreferencesSetMenuItemElementEnabled and
WKPreferencesGetMenuItemElementEnabled in order to preserve binary
compatibility.

Due to file removal a couple unified build fixes were needed. Those are
annotated inline below.

* LayoutTests/fast/html/menuitem-element-expected.txt: Removed.
* LayoutTests/fast/html/menuitem-element.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/html/HTMLMenuElement.cpp:
(WebCore::HTMLMenuElement::insertedIntoAncestor): Deleted.
(WebCore::HTMLMenuElement::removedFromAncestor): Deleted.
(WebCore::HTMLMenuElement::attributeChanged): Deleted.
* Source/WebCore/html/HTMLMenuElement.h:
* Source/WebCore/html/HTMLMenuItemElement.cpp: Removed.
* Source/WebCore/html/HTMLMenuItemElement.h: Removed.
* Source/WebCore/html/HTMLMenuItemElement.idl: Removed.
* Source/WebCore/html/HTMLTagNames.in:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::didInsertMenuElement): Deleted.
(WebCore::ChromeClient::didRemoveMenuElement): Deleted.
(WebCore::ChromeClient::didInsertMenuItemElement): Deleted.
(WebCore::ChromeClient::didRemoveMenuItemElement): Deleted.
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.h:

Unified build fix.

* Source/WebKit/Shared/TouchBarMenuData.cpp: Removed.
* Source/WebKit/Shared/TouchBarMenuData.h: Removed.
* Source/WebKit/Shared/TouchBarMenuItemData.cpp: Removed.
* Source/WebKit/Shared/TouchBarMenuItemData.h: Removed.
* Source/WebKit/Shared/TouchBarMenuItemData.serialization.in: Removed.
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetMenuItemElementEnabled): Deleted.
(WKPreferencesGetMenuItemElementEnabled): Deleted.
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::touchBarMenuDataChanged): Deleted.
(WebKit::WebPageProxy::touchBarMenuItemDataAdded): Deleted.
(WebKit::WebPageProxy::touchBarMenuItemDataRemoved): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::didInsertMenuElement): Deleted.
(WebKit::WebChromeClient::didRemoveMenuElement): Deleted.
(WebKit::WebChromeClient::didInsertMenuItemElement): Deleted.
(WebKit::WebChromeClient::didRemoveMenuItemElement): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didInsertMenuElement): Deleted.
(WebKit::WebPage::didRemoveMenuElement): Deleted.
(WebKit::WebPage::didInsertMenuItemElement): Deleted.
(WebKit::WebPage::didRemoveMenuItemElement): Deleted.
(WebKit::WebPage::sendTouchBarMenuDataRemovedUpdate): Deleted.
(WebKit::WebPage::sendTouchBarMenuDataAddedUpdate): Deleted.
(WebKit::WebPage::sendTouchBarMenuItemDataAddedUpdate): Deleted.
(WebKit::WebPage::sendTouchBarMenuItemDataRemovedUpdate): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:

Includes unified build fix.

* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences menuItemElementEnabled]): Deleted.
(-[WebPreferences setMenuItemElementEnabled:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebPreferencesPrivate.h:

Canonical link: https://commits.webkit.org/267140@main
